src/main/java/pages/FlightSearchPage.java

```diff
@@ -0,0 +1,223 @@
+    private WebElement departureInput;
+
+    @FindBy(id = "booking")
+    private WebElement searchButton;
+
+    @FindBy(id = "one-way-toggler")
+    private WebElement oneWayToggleCheckbox;
+
+    @FindBy(css = "label.input-toggler")
+    private WebElement toggleLabel;
+
+    public FlightSearchPage(WebDriver driver) {
+        this.driver = driver;
+        this.wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        PageFactory.initElements(driver, this);
+    }
+
+    public void enableRoundTripIfDisabled() {
+        wait.until(ExpectedConditions.elementToBeClickable(toggleLabel));
+        if (oneWayToggleCheckbox.isSelected()) {
+            toggleLabel.click();
+            wait.until(d -> {
+                try {
+                    WebElement returnInput = d.findElement(By.cssSelector("input[data-note]"));
+                    return returnInput.isDisplayed();
+                } catch (Exception e) {
+                    return false;
+                }
+            });
+        }
+    }
+
+    public void selectFromCity(String cityName) {
+        fromButton.click();
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("ul.dropdown-menu.inner.show")));
+        WebElement city = wait.until(ExpectedConditions.elementToBeClickable(
+                By.xpath("//ul[contains(@class,'dropdown-menu')]//span[contains(text(), '" + cityName + "')]")));
+        scrollIntoView(city);
+        city.click();
+    }
+
+    public void selectToCity(String cityName) {
+        wait.until(ExpectedConditions.elementToBeClickable(toButton)).click();
+        try {
+            WebElement searchBox = wait.until(ExpectedConditions.presenceOfElementLocated(
+                    By.cssSelector("div.dropdown-menu.show input.form-control")));
+            searchBox.clear();
+            searchBox.sendKeys(cityName);
+            Thread.sleep(1000);
+            searchBox.sendKeys(Keys.ENTER);
+        } catch (TimeoutException e) {
+            WebElement city = wait.until(ExpectedConditions.elementToBeClickable(
+                    By.xpath("//ul[contains(@class,'dropdown-menu')]//span[contains(text(), '" + cityName + "')]")));
+            scrollIntoView(city);
+            city.click();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    public void selectDepartureDate(String day) {
+        ((JavascriptExecutor) driver).executeScript("arguments[0].click();", departureInput);
+
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".drp-calendar")));
+
+        while (!isTargetMonthVisible("July 2025")) {
+            List<WebElement> nextButtons = driver.findElements(By.cssSelector(".next.available"));
+            if (nextButtons.isEmpty()) {
+                throw new NoSuchElementException("No .next.available button found — cannot navigate to July 2025");
+            }
+            nextButtons.get(0).click();
```

Comment:
Let's avoid getting elements by index, it might lead to test instability as elements order might change. 

Why do we have several nextButtons on the page? is it possible to create a more specific locator?